### PR TITLE
PR for BXMSDOC-7489: Update and single-source the Building custom dashboard widgets doc based on BAPL-1863, BAPL-1865, and BAPL-1862

### PR DESCRIPTION
**Pull request by kaldesai**
Associated JIRA: https://issues.redhat.com/browse/BXMSDOC-7489

**Affected BAPLs for the change:**
- [BAPL-1862](https://issues.redhat.com/browse/BAPL-1862): [Dashbuilder] Prometheus Data Set Provider
- [BAPL-1863](https://issues.redhat.com/browse/BAPL-1863): [Dashbuilder] Kafka Data Set Provider
- [BAPL-1865](https://issues.redhat.com/browse/BAPL-1865): [Dashbuilder] Time Series Displayer

**Enterprise document previews:**
- [RHPAM-IV. Building custom dashboard widgets](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-7489-RHPAM-time-series/#proc-adding-time-series-components_building-custom-dashboard-widgets)
- [RHDM-IV. Building custom dashboard widgets](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-7498-RHDM-TimeSeries/#proc-creating-time-series-components_building-custom-dashboard-widgets) 

**Doc impact for enterprise documents:**
- Check section **54.4.7. Creating time series component** in RHPAM doc.

**Community document previews:**
- [jBPM document](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-7489-TS-jBPM/#proc-adding-time-series-components_business-processes)

**Doc impact for community documents:**
- Check **22.2.10. Creating time series component** in jBPM doc.

Can you please review and verify the community doc and share your feedback?

### CODE REVIEW

**Review by manaswinidas — CHANGES_REQUESTED** [review]
Hi @kaldesai , I've left some comments, please check if they make sense.

File: doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+|You can select the *zoom*, *selection*, or *pan* option from the drop-down list as per your requirements.
+
+|*Title text*
+|You can edit the title of the time series component.
+
+|*Title align*
+|You can select the *left*, *center*, or *right* option from the drop-down list as per your requirements to change the title alignment .
+
+|*Show toolbar*
+|It is a checkbox. By default it is selected. If this component property is selected, you can use the zoom in, zoom out, selection zoom, panning functions of the time series component.
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc:72]:
```suggestion
|Select the checkbox to display the toolbar. By default, this checkbox is checked. If this component property is enabled, you can use the zoom in, zoom out, selection zoom, and panning functions of the time series chart component.
```

File: doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+|*Chart Name*
+|You can set the chart name as per your requirements. By default it is set to *Newchart*.
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc:75]:
```suggestion
|Set the chart name. By default, the chart name is set as *Newchart*.
```

**Review by hmanwani-rh — COMMENTED** [review]
@kaldesai Please check the comments and suggestions below and let me know if you have any questions.

File: doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc:72]: Done.

File: doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+|It is a checkbox. You can enable or disable data labels on data points.
+
+|*Zoom type*
+|You can select the *x*, *y*, or *xy* option from the drop-down list as per your requirements.
+
+|*Zoom enabled*
+|It is a checkbox. By default it is selected.
+
+|*Zoom autoscale Y axis*
+|It is a checkbox. You can zoom in the Y axis.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc:60]: Added.

File: doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+|You can select the *category*, *datetime*, or *numeric* option from the drop-down list as per your requirements.
+
+|*Labels*
+|It is a checkbox. You can enable or disable data labels on data points.
+
+|*Zoom type*
+|You can select the *x*, *y*, or *xy* option from the drop-down list as per your requirements.
+
+|*Zoom enabled*
+|It is a checkbox. By default it is selected.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc:57]: Added

File: doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+[cols="1,1", options="header"]
+|===
+| Component properties
+| Description
+
+|*Transpose dataset*
+|You can decide whether he provided data set uses the time series as separate columns or as rows.
+
+|*Show Area*
+|It is a checkbox. You can set the type as a chart, area, or line.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc:45]: Updated

File: doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc [hunk 6]
```diff
@@ -0,0 +1,83 @@
+[id='proc-creating-time-series-components_{context}']
+
+= Creating time series component
+
+You can use the time series component to represent your data in the time series data. You can create your own dashboards connected to your time series data sets.
+
+You can export the time series components to Dashbuilder Runtime and retrieve the information from a KIE Server data set. You can also create, edit, and build a dashboard using a time series component.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc:7]: Updated

File: doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc [hunk 7]
```diff
@@ -0,0 +1,83 @@
+[id='proc-creating-time-series-components_{context}']
+
+= Creating time series component
+
+You can use the time series component to represent your data in the time series data. You can create your own dashboards connected to your time series data sets.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc:5]: Updated.

File: doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc [hunk 8]
```diff
@@ -0,0 +1,83 @@
+[id='proc-creating-time-series-components_{context}']
+
+= Creating time series component
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/jbpm-docs/src/main/asciidoc/BAM/proc-creating-time-series-components.adoc:3]: Yes, +1 to that. I will change the title of the procedure.

File: doc-content/jbpm-docs/src/main/asciidoc/BAM/con-building-custom-dashboard-reporting-components.adoc [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+|This component is used to display the data from data sets in the form of metric. You can use the *Preview*, *HTML*, or *Javascript* tab to edit your data.
+
+|*Table*
+|This component is used to display the data from data sets in the form of table. You can hide or display the columns as per your requirements.
+
+|*Filter*
+|This component allows you to filter the data from data sets. 
+
+|*Time Series Chart*
+|This component is used to display the data from data sets in the form time series.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/jbpm-docs/src/main/asciidoc/BAM/con-building-custom-dashboard-reporting-components.adoc:49]: Yes, that's true. 
Typo :|

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-process-heatmap-components-proc.adoc [hunk 10]
```diff
@@ -31,37 +30,26 @@ The *Data Set Creation Wizard* page opens.
...
      pil.externalId,
-     nid,
-     nodename
+     nil.nodeid,
+     nil.nodename
 ----
++
+[NOTE]
+====
+You can modify the SQL query according to your database as per the requirements.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-process-heatmap-components-proc.adoc:51]: Done

File: doc-content/jbpm-docs/src/main/asciidoc/BAM/con-building-custom-dashboard-components.adoc [hunk 11]
```diff
@@ -0,0 +1,11 @@
+[id='con-building-custom-dashboard-components_{context}']
+= Components
+
+A page contains different types of components. From the *Components* panel, you can use the following component types in a page:
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/jbpm-docs/src/main/asciidoc/BAM/con-building-custom-dashboard-components.adoc:4]: Updated

**Review by manaswinidas — APPROVED** [review]

File: doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-all-processes-heatmap-components-proc.adoc [hunk 12]
```diff
@@ -61,7 +61,7 @@ You can modify the SQL query according to your database if required.
 The new page opens in the *Page Editor* and you can see that the *Heatmaps* component is available under the *Components* panel.
 . In the *Components* panel, expand the *Heatmaps* components and drag the *All Processes Heatmaps* component type to the canvas.
 . In the *Displayer editor* wizard, click *Data* tab and select the newly created {KIE_SERVER} dataset.
-. In the *Data* tab, select *EXTERNALID*, *PROCESSID*, *NID*, and *AVERAGEEXECUTIONTIME* from the *Columns* field.
+. In the *Data* tab, select *NODEID* and *TOTAL_HITS* from the *Columns* field.
```
Comment (sthundat) [inline doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-all-processes-heatmap-components-proc.adoc:64]: Hi @kaldesai ,
for all process heapmap column values are EXTERNALID,PROCESSID,NODEID and TOTAL_HITS

**Pull request merged**